pkgs/repo_manage/lib/issue_transfer.dart

```diff
@@ -0,0 +1,227 @@
+      }
+    }
+
+    var parsedData =
+        await transferIssue(sourceRepo, targetRepo, labelName, applyChanges);
+
+    if (labelName != null) {
+      for (var issueNumber in parsedData) {
+        print('Add label $labelName to issue $issueNumber');
+        if (applyChanges) {
+          await reportRunner.github.issues.addLabelsToIssue(
+            targetRepo,
+            issueNumber,
+            [labelName],
+          );
+        }
+      }
+    }
+
+    return 0;
+  }
+
+  Future<List<String>> getIssueIds(
+    RepositorySlug slug, [
+    List<int>? issueNumbers,
+  ]) async {
+    final queryString = '''query {
+  repository(owner:"${slug.owner}", name:"${slug.name}") {
+    issues(last:100) {
+      nodes {
+        id
+        number
+      }
+    }
+  }
+}
+''';
+    final result = await query(QueryOptions(
+      document: gql(queryString),
+      // If the cache is enabled this will always return the same issues, even
+      // after transferring them to another repo.
+      fetchPolicy: FetchPolicy.noCache,
+      parserFn: (data) {
+        var repository = data['repository'] as Map;
+        var issues = repository['issues'] as Map;
+        var nodes = issues['nodes'] as List;
+        return nodes
+            .map((node) => node as Map)
+            .where((node) => issueNumbers != null
+                ? issueNumbers.contains(node['number'] as int)
+                : true)
+            .map((node) => node['id'] as String)
+            .toList();
+      },
+    ));
+
+    return result.hasException ? throw result.exception! : result.parsedData!;
+  }
+
+  Future<String> getRepositoryId(RepositorySlug slug) async {
+    final queryString = '''query {
+  repository(owner:"${slug.owner}", name:"${slug.name}") {
+    id
+  }
+}
+''';
+    final result = await query(QueryOptions(
+      document: gql(queryString),
+      parserFn: (data) {
+        var repository = data['repository'] as Map;
+        return repository['id'] as String;
+      },
+    ));
+
+    return result.hasException ? throw result.exception! : result.parsedData!;
+  }
+
+  Future<List<int>> transferIssue(
+    RepositorySlug sourceRepo,
+    RepositorySlug targetRepo,
+    String? issueLabel,
+    bool applyChanges,
+  ) async {
+    var repositoryId = await getRepositoryId(targetRepo);
+    var allIssueIds = <int>[];
+    while (true) {
+      var issueIds = await getIssueIds(sourceRepo);
+      if (issueIds.isEmpty) {
+        print('Done transferring a total of ${allIssueIds.length} issues from '
+            '$sourceRepo to $targetRepo');
+        return allIssueIds;
+      }
+      print('Transfer ${issueIds.length} issues from $sourceRepo to $targetRepo'
+          'with id $repositoryId');
+      var transferredIssues =
+          await _transferMutation(issueIds, repositoryId, applyChanges);
+      allIssueIds.addAll(transferredIssues);
+      if (!applyChanges) {
+        return List.generate(issueIds.length, (index) => index);
```

Comment:
Hmm, this is returning a mock list of issue ids if !applyChanges? Is that useful? Perhaps return an empty list?

Reply:
Or you want some explicit mock text to print out when in dry run mode? `'Add label $labelName to issue $issueNumber'`

Reply:
Yeah, the idea was to see how the workflow would continue if the issues would have been transferred...